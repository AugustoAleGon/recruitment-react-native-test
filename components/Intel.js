import React from 'react'
import { View, Text, StyleSheet, Button } from 'react-native'

const IntelListItem = props => (
  <View style={styles.intelItem}>
    <Text style={{ justifyContent: 'center' }}>
      <Text style={styles.intelProduct}>
        {props.product}
        {'  '}
      </Text>
      <Text style={styles.intelDate}>{props.date}</Text>
    </Text>
    <View>
      <Button
        style={styles.intelTitle}
        title={props.title}
        onPress={() => props.navigation.navigate('IntelViewer', { ...props })}
      >
        {'\n'}
        {props.title}
        {'\n'}
      </Button>
    </View>
    <Text>
      <Text style={styles.intelProduct}>
        {props.location}
        {'  '}
      </Text>
      <Text style={styles.intelDate}>{props.tag}</Text>
    </Text>
  </View>
)

const styles = StyleSheet.create({
  intelItem: {
    flex: 1,
    height: '100%',
    backgroundColor: '#e8e8e8',
    justifyContent: 'center',
    padding: 10,
    borderBottomWidth: 1,
    borderBottomColor: 'black'
  },
  intelProduct: {
    flex: 1,
    fontSize: 10,
    fontWeight: 'bold',
    width: 25
  },
  intelTitle: {
    flex: 1,
    fontWeight: 'bold',
    fontSize: 14,
    alignItems: 'flex-start'
  },
  intelDate: {
    fontWeight: 'normal',
    marginLeft: 10,
    flex: 1,
    fontSize: 10,
    alignItems: 'center'
  }
})

export default IntelListItem
