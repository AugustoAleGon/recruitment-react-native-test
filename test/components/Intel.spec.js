import React from 'react'
import renderer from 'react-test-renderer'
import Intel from '../../components/Intel'
import sinon from 'sinon'

describe('Intel', () => {
  it('should render the text of the header', () => {
    const props = {
      product: 'Test product',
      title: 'Test title',
      location: 'London',
      tag: 'Test tag',
      navigation: {
        navigate: sinon.spy()
      }
    }
    const wrapper = renderer.create(<Intel {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
