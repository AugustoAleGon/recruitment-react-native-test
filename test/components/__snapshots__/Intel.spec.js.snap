// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intel should render the text of the header 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e8e8e8",
      "borderBottomColor": "black",
      "borderBottomWidth": 1,
      "flex": 1,
      "height": "100%",
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "flex": 1,
          "fontSize": 10,
          "fontWeight": "bold",
          "width": 25,
        }
      }
    >
      Test product
        
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "fontSize": 10,
          "fontWeight": "normal",
          "marginLeft": 10,
        }
      }
    />
  </Text>
  <View>
    <View
      accessibilityComponentType="button"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Test title
        </Text>
      </View>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "flex": 1,
          "fontSize": 10,
          "fontWeight": "bold",
          "width": 25,
        }
      }
    >
      London
        
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "fontSize": 10,
          "fontWeight": "normal",
          "marginLeft": 10,
        }
      }
    >
      Test tag
    </Text>
  </Text>
</View>
`;
